indico/modules/events/registration/client/js/form/fields/SingleChoiceInput.jsx

```diff
@@ -68,73 +79,85 @@ function SingleChoiceDropdown({
             (placesUsed[selectedChoice.id] || 0) - (existingValue[selectedChoice.id] || 0) >=
               selectedChoice.placesLimit)
         }
-        value={value[selectedChoice.id]}
-        onChange={(_evt, data) => onChange({[selectedChoice.id]: data.value})}
-        fluid
+        value={String(selectedSeats)}
```

Comment:
When changing between two options there's a bug when 'extra slots' are enabled:

- The combobox value is not cleared, so if I switch from a max 3 slots choice (w/ 3 selected) to one that allows max 2 slots, the combobox still shows "3"
- Regardless of the choice's validity, when changing the option the combobox keeps the displayed value, which is now different from the submitted value. So If I change the choice like I mentioned in my previous point, it looks like I'm submitting the choice with 3 slots, but in reality I submit it with 1 slot.

Reply:
Will check

Reply:
Likely related: When you use the `x` icon to clear the selected choice, the slots dropdown does not disappear.